tasks/deploy.js

```diff
@@ -0,0 +1,249 @@
+    const network = hre.hardhatArguments.network;
+    console.log(`Deploying tribute-contracts to ${network} network`);
+
+    const { contracts } = require(`../deployment/${network}.config`);
+    const hardHatImports = require("../utils/HardhatUtil.js")(contracts);
+    const DaoArtifacts = hardHatImports.DaoArtifacts;
+    
+    let daoArtifacts;
+    
+    if (process.env.DAO_ARTIFACTS_CONTRACT_ADDR) {
+        console.log(`Attach to existing DaoArtifacts contract`);
+        daoArtifacts = await DaoArtifacts.attatch(
+          process.env.DAO_ARTIFACTS_CONTRACT_ADDR
+        );
+    } else {
+        console.log(`Creating new DaoArtifacts contract`);
+        const daoArtifact = await (await DaoArtifacts).deploy();
+        daoArtifacts = await daoArtifact.deployed();
+    }
+
+    console.log(`DaoArtifacts: ${daoArtifacts.address}`);
+  
+    const deployFunction = hardHatImports.deployFunctionFactory(
+      hre,
+      daoArtifacts
+    );
+
+    if (network === "mainnet") {
+        res = await deployMainnetDao(deployFunction, network, hardHatImports);
+    } else if (network === "ganache") {
+        const accounts = await hre.ethers.getSigners();
+        res = await deployGanacheDao(
+          deployFunction,
+          network,
+          accounts,
+          hardHatImports
+        );
+    } else if (network === "rinkeby") {
+        res = await deployRinkebyDao(deployFunction, network, hardHatImports);
+    } else if (network === "test" || network === "coverage") {
+        res = await deployTestDao(
+          deployFunction,
+          network,
+          accounts,
+          hardHatImports
+        );
+    }
+
+    let { dao, extensions, testContracts } = res;
+
+    if (dao) {
+        await dao.finalizeDao();
+        console.log("************************");
+        console.log(`DaoRegistry: ${dao.address}`);
+        console.log(
+          `Multicall: ${
+            testContracts.multicall ? testContracts.multicall.address : ""
+          }`
+        );
+        console.log(`BankExtension: ${extensions.bank.address}`);
+        console.log(
+          `NFTExtension: ${extensions.nft ? extensions.nft.address : ""}`
+        );
+        console.log(
+          `ERC20Extension: ${
+            extensions.erc20Ext ? extensions.erc20Ext.address : ""
+          }`
+        );
+        console.log("************************");
+    } else {
+        console.log("************************");
+        console.log("no migration for network " + network);
+        console.log("************************");
+    }
+});
+
+const deployRinkebyDao = async (deployFunction, network, hardHatImports) => {
+    if (!process.env.DAO_NAME)
+        throw Error("Missing env var: DAO_NAME");
+    if (!process.env.DAO_OWNER_ADDR)
+        throw Error("Missing env var: DAO_OWNER_ADDR");
+    if (!process.env.ERC20_TOKEN_NAME)
+        throw Error("Missing env var: ERC20_TOKEN_NAME");
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/openlawteam/tribute-contracts/pullRequest?prid=8819146)
```suggestion
    {throw Error("Missing env var: ERC20_TOKEN_NAME");}
```